src/main/java/org/wso2/carbon/inbound/snmp/SNMPListeningConsumer.java

```diff
@@ -0,0 +1,282 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * SNMP inbound endpoint is used to listen and consume messages from SNMP agent via WSO2 ESB.
+ */
+public class SNMPListeningConsumer extends GenericEventBasedConsumer implements CommandResponder {
+
+    private static final Log logger = LogFactory.getLog(SNMPListeningConsumer.class);
+    private Address listenAddress;
+    private Snmp snmp = null;
+    private ThreadPool threadPool;
+    private String verifiedSnmpVersion;
+
+    /**
+     * @param properties            a persistent set of properties for InboundEndpoint.
+     * @param name                  name of the inbound endpoint
+     * @param synapseEnvironment    synapse environment
+     * @param injectingSeq          name of the sequence message that should be injected
+     * @param onErrorSeq            name of the fault sequence that should be invoked in case of failure
+     * @param coordination          this parameter is only applicable in a cluster environment
+     * @param sequential            The behavior when executing the given sequence.
+     */
+    public SNMPListeningConsumer(Properties properties, String name, SynapseEnvironment synapseEnvironment,
+                                 String injectingSeq, String onErrorSeq, boolean coordination, boolean sequential) {
+
+        super(properties, name, synapseEnvironment, injectingSeq, onErrorSeq, coordination, sequential);
+        logger.info("Starting to load the SNMP Inbound Endpoint " + name);
+
+        String host = properties.getProperty(SNMPConstants.HOST);
+        int port;
+        String snmpVersion = properties.getProperty(SNMPConstants.SNMP_VERSION);
+        boolean isTCP = Boolean.parseBoolean(properties.getProperty(SNMPConstants.IS_TCP));
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("Starting to load the SNMP Properties for " + name);
+        }
+
+        if (StringUtils.isEmpty(host)) {
+            throw new SynapseException("IP address of the SNMP is not set");
+        }
+
+        if (StringUtils.isEmpty(properties.getProperty(SNMPConstants.PORT))) {
+            throw new SynapseException("Port to access the " + host + " is not set");
+        } else {
+            port = Integer.parseInt(properties.getProperty(SNMPConstants.PORT));
+        }
+
+        if (snmpVersion.equals(SNMPConstants.SNMP_VERSION2C) || snmpVersion.equals(SNMPConstants.SNMP_VERSION1)) {
+            this.verifiedSnmpVersion = snmpVersion;
+        } else if (StringUtils.isEmpty(snmpVersion)) {
+            this.verifiedSnmpVersion = SNMPConstants.SNMP_VERSION2C;
+        } else {
+            if (logger.isDebugEnabled()) {
+                logger.debug("SNMP inbound doesn't support SNMP version other than 2c and 1");
+            }
+            throw new SynapseException("SNMP connector doesn't support SNMP version other than 1 and 2c");
```

Comment:
move this message "SNMP inbound doesn't support SNMP version other than 2c and 1" to constant and use it in all places